dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -67,14 +67,17 @@ class VectorRange {
 };
 
 struct ComponentDesc {
-  std::string ext;
+  std::string filename, ext;
   size_t size = 0;
   int64_t offset = 0;
   VectorRange<size_t> outputs;
 
   ComponentDesc() = default;
-  ComponentDesc(std::string new_ext, size_t new_size, int64_t new_offset)
-      : ext(std::move(new_ext)), size(new_size), offset(new_offset) {}
+  ComponentDesc(std::string new_filename, std::string new_ext, size_t new_size, int64_t new_offset)
```

Comment:
```suggestion
  ComponentDesc(std::string key, std::string ext, size_t size, int64_t offset)
```
The `new` suggests that it will overwrite something existing which it doesn't. Also, this entire constructor is not really necessary - a brace-enclosed initializer would do fine.

Reply:
Removed the ctor